spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -530,14 +515,16 @@ private static void importUnpartitionedSparkTable(
    * @param spec a partition spec
    * @param stagingDir a staging directory to store temporary manifest files
    * @param checkDuplicateFiles if true, throw exception if import results in a duplicate data file
+   * @param listingParallelism the parallelism to use when listing files
    */
   public static void importSparkPartitions(
       SparkSession spark,
       List<SparkPartition> partitions,
       Table targetTable,
       PartitionSpec spec,
       String stagingDir,
-      boolean checkDuplicateFiles) {
+      boolean checkDuplicateFiles,
+      int listingParallelism) {
```

Comment:
Important: This is a public utility, we can't just add a parameter without breaking users who may be using this (and this API has existed for a while) . It's better to duplicate and add a new a method (and then leverage the implementation of the parallel one in the original one)